src/tooltip/Tooltip.tsx

```diff
@@ -1,357 +1,223 @@
-      return this.renderStaticModalContent();
-    } else {
-      return this.renderTogglingModalContent();
+  React.useEffect(() => {
+    Animated.timing(animation, {
+      toValue: Number(visible),
+      useNativeDriver: true,
+      duration: 200,
+    }).start();
+  }, [visible, animation]);
+
+  React.useEffect(() => {
+    Dimensions.addEventListener('change', calcDim);
+    return () => {
+      Dimensions.removeEventListener('change', calcDim);
+    };
+  }, []);
+
+  React.useEffect(() => {
+    calcDim();
+  }, [visible]);
+
+  const ToolTipStyle = () => {
+    const { px, width, height, py } = containerDimensions;
+    const { width: tooltipWidth, height: toolTipHeight } = tooltipDimension;
+    let horizontal =
+      (px || 0) +
+      ((px || 0) > ScreenWidth / 2
+        ? ((width || 0) - (tooltipWidth || 0)) / 2
+        : 20) +
+      (width > ScreenWidth / 2
+        ? (width || 0) / 2 - (tooltipWidth || 0) / 2
+        : 0) +
+      (px < 10 ? 10 : 0) +
+      (width < 30 ? -20 : 0);
+    if (horizontal > ScreenWidth - tooltipWidth) {
+      horizontal = ScreenWidth - (tooltipWidth || 0);
     }
+    let top = py + (ScreenHeight / 2 < py ? -toolTipHeight - height - 28 : 0);
+    return { top, [I18nManager.isRTL ? 'right' : 'left']: horizontal };
   };
 
-  render() {
-    const { isVisible } = this.state;
-    const { onOpen, ModalComponent } = this.props;
-    return (
+  return (
+    <>
       <View
-        collapsable={false}
-        ref={(e) => {
-          this.renderedElement = e;
-        }}
+        style={styles.content}
+        ref={containerElement}
+        onLayout={(e) => Object(e.nativeEvent.layout)}
       >
-        {this.renderContent(false)}
-        <ModalComponent
-          animationType="fade"
-          visible={isVisible}
-          transparent
-          onShow={onOpen}
+        <Pressable
+          {...(!closeOnlyOnBackdropPress && { [toggleAction]: toggleToolTip })}
         >
-          {this.renderModalContent()}
-        </ModalComponent>
+          {children}
+        </Pressable>
+        <Modal transparent visible={visible}>
+          <TouchableWithoutFeedback onPress={toggleToolTip}>
+            <Animated.View
+              style={[
+                StyleSheet.absoluteFillObject,
+                {
+                  backgroundColor: overlayColor || '#b4b0b087',
+                  opacity: animation,
+                },
+              ]}
+              pointerEvents={visible ? 'auto' : 'none'}
+            />
+          </TouchableWithoutFeedback>
+          <SafeAreaView>
+            <Animated.View
+              pointerEvents={visible ? 'auto' : 'none'}
+              style={[
+                styles.tooltip,
+                {
+                  opacity: animation,
+                },
+              ]}
+            >
+              {withPointer ? (
+                <Triangle
+                  isDown={ScreenHeight / 2 < containerDimensions.py}
```

Comment:
`position` prop should be defined by the user 
Something like this -> https://material-ui.com/components/tooltips/#positioned-tooltips

Reply:
Like this?
![A](https://raw.githubusercontent.com/doomsower/react-native-modal-popover/master/gifs/android.gif)